src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
-    </p>
-  </form>
-);
+import React, { ChangeEvent, useState } from "react";
+import { Filter } from "../../App";
+
+interface Props {
+  filterBy: (filters: Filter) => void,
+}
+
+
+export const TodoFilter: React.FC<Props> = ({ filterBy }) => {
+  const [status, setStatus] = useState('all');
+  const [title, setTitle] = useState('');
+
+  const handleStatusChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setStatus(event.target.value);
+    filterBy({ status: event.target.value, title: title });
+  };
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    filterBy({ status: status, title: event.target.value });
+  };
+
+  const onClear = () => {
+    setStatus('all');
+    setTitle('');
+    filterBy({ status: 'all', title: '' });
+  };
+
+  return (
+    <>
+      <form className="field has-addons">
+        <p className="control">
+          <span className="select">
+            <select
+              data-cy="statusSelect"
+              value={status}
+              onChange={handleStatusChange}
+            >
+              <option value="all">All</option>
+              <option value="active">Active</option>
+              <option value="completed">Completed</option>
+            </select>
+          </span>
+        </p>
+
+        <p className="control is-expanded has-icons-left has-icons-right">
+          <input
+            data-cy="searchInput"
+            type="text"
+            className="input"
+            placeholder="Search..."
+            value={title}
+            onChange={handleInputChange}
+          />
+          <span className="icon is-left">
+            <i className="fas fa-magnifying-glass" />
+          </span>
+
+          {title && (
```

Comment:
```suggestion
          {!!title && (
```
in rendering with && for best rendering control use !! to return a boolean value